docs/1-introduction/1-3-models/1-3-1-texturing.md

```diff
@@ -0,0 +1,162 @@
+# Texturing
+
+[Project on GitHub](https://github.com/GraphicsProgramming/learnd3d11/tree/main/src/Cpp/1-getting-started/1-3-1-Texturing)
+
+Now that we have a triangle, the next step is getting a texture onto that triangle. 
+However in order to get there we need to take a few steps.
+Textures are usually created from images, however these images almost always come in some compressed form.
+We cannot just simply read in a PNG file and create a texture from it, the GPU does not understand the PNG format (or most other formats) with the exception of a few special formats.
+
+Thus in order to create a texture from an image we'll have to decode it, this can be a very difficult process so we will use a library for it, in our case two even!
+We'll quickly explain both FreeImage and DirectXTex here but you can use whatever you are comfortable with in the end, we have chosen these two libraries because they both do specific things.
+
+- "FreeImage" will allow us to read nearly every common format (PNG,JPG,TGA,BMP, etc) and give us a few options to preprocess them.
+- "DirectXTex" will allow us to read in the DDS format, which will be explained in a bit.
+
+Let's take a look at the code, we'll mainly focus on two functions here, `CreateTextureView` and `CreateTextureViewFromDDS`.
+
+Firstly, `CreateTextureView`:
+In this function we'll use FreeImage to read in regular images and create a texture from it. After reading in the full file `CreateFileW()` -> `ReadFile()` we tell FreeImage to open a handle to the raw image data (which is still encoded) with: `FreeImage_OpenMemory(fileDataRaw.data(), fileDataRaw.size())`.
```

Comment:
Can you turn `CreateFileW` and `ReadFile` into hyperlinks to winapi docs?

And if there is for FreeImage/DirectXTex the same when calling their functions, people can look at their docs right away